livekit-agents/livekit/agents/pipeline/pipeline_agent.py

```diff
@@ -660,6 +754,19 @@ async def _synthesize_answer_task(
             SpeechDataContextVar.reset(tk)
 
     async def _play_speech(self, speech_handle: SpeechHandle) -> None:
+        if speech_handle.extra_tools_messages:
+            # synthesize the tool speech with the latest chat context and the tools messages
```

Comment:
I moved the llm call for the tool outputs here to be able to use the latest chat context. It's intended to avoid repeating the same sentence from `say` before the function call completed. Let me know if this makes sense.
```
user: What is the weather in New York?
agent (say): Let me check that for you.
(function call...)
agent: The weather in New York is ...
```

Reply:
Could you optimize and check if the speech is interruptable? The way I'm thinking of this - it'll just be a short interjection ("Thanks for sharing!") so I'm fine making that uninterruptable.

Reply:
Since we are now running the function call asynchronously, there are three options when human input occurs while the function call is still running:
1. ignore the human input and while the function calls finishing (the existing way)
2. interrupt the function calls
3. answer the human input meanwhile keep function calls and insert the function call answer later

I have discussed with @theomonnom that I'll make all function calls uninterruptable to have the same behavior with the existing version (option 1). And maybe in the future we can make this configurable and the developer can choose which behavior to have.

(Note that the above is only for human input, the speech from `say` will always be send out when the function call is running.)